projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -3219,6 +3219,9 @@ export abstract class IgxGridBaseDirective extends DisplayDensityBase implements
             Object.keys(obj).forEach(key => isObject(obj[key]) ? changes += f(obj[key]) : changes++);
             return changes;
         };
+        if (this.transactions.getState(this.crudService.row.id)?.type === TransactionType.ADD) {
+            return this._columns.filter(c => c.field).length;
```

Comment:
Are there columns without a field and why are they excluded from the count?

Reply:
The field property of the unbound columns is undefined and there was a meeting where we have discussed to include only the defined columns which are included in the data.

Reply:
@MonikaKirkova OK, makes sense. However, I'm not finding anything in the docs about `unbound column`. I didn't know we support this, nor can I find that it's done without specifying a field property value.

Reply:
@MonikaKirkova I know what it is :) I was making the comment for @ChronosSF, because us, as developers of the product - we know how to create something like an unbound column, but developers who consume the library usually search for this in the docs!

Reply:
Logged https://github.com/IgniteUI/igniteui-docfx/issues/3388 to address this